freqtrade/persistence/models.py

```diff
@@ -232,7 +236,7 @@ class LocalTrade():
     close_rate_requested: Optional[float] = None
     close_profit: Optional[float] = None
     close_profit_abs: Optional[float] = None
-    stake_amount: float = 0.0
+    stake_amount: float = 0.0  # TODO: This should probably be computed
```

Comment:
I'm not sure i understand this TODO.

stake-amount is the value (in stake-amount) that you invest.
so assuming a XRP/USDT trade - you buy XRP for 1000$ - so 1000$ is your stake-amount.
This is irrelevant of side (you might buy for 1000$, or sell for 1000$) - and also irrelevant of leverage (leverage will determine the amount of XRP you get).
In it's current form - it will also not change after the trade has been entered - so having it computed does not provide any visible benefit.

Moving this to "computed" will be problematic for other reasons, too, as it's a very important measure for "post-mortem" analysis - and as such - should be in the database.

Reply:
It doesn't have to be. I did run into some problems caused by it in the `borrowed` variable earlier when it was

```
elif not self.is_short:
            return self.stake_amount * ((self.leverage-1)/self.leverage)
```

and I was changing the amount within my tests, which is why I updated this calculation to

```
elif not self.is_short:
            return (self.amount * self.open_rate) * ((self.leverage-1)/self.leverage)
```

I cant remember exactly what the problem was, but I remember the values not being synced

Reply:
well it might be a testing problem (where you change one thing over the other between different asserts)
i'd not take that assumption and apply it to the real world (as said, stake-amount is the value you invest - not some "random" value that's calculated). You paid for that - exactly that ...